types/src/proof/mod.rs

```diff
@@ -2,6 +2,12 @@
 // SPDX-License-Identifier: Apache-2.0
 
 pub mod accumulator;
+pub mod proof_error;
+// This module actually needs and uses the imports that it complains about not using
+// (SparseMerkleProof and ProofError)
+#[allow(unused_imports)]
```

Comment:
I don't think you need this, I think it's complaining cause you made it pub mod when you only use it in types crate

Reply:
I tried removing this and making the module private, but still got these clippy errors:
<img width="503" alt="Screen Shot 2019-11-07 at 3 18 57 PM" src="https://user-images.githubusercontent.com/22864715/68436124-ee598d80-0171-11ea-8894-8387203d02e5.png">

I think this is a similar situation to this: https://github.com/libra/libra/blob/master/common/logger/src/lib.rs#L17-L23
If there's another workaround, please share! :)